mmgen/models/architectures/biggan/generator_discriminator.py

```diff
@@ -0,0 +1,532 @@
+        input_scale (int, optional): [description]. Defaults to 4.
+        with_shared_embedding (bool, optional): [description]. Defaults to
+            True.
+        shared_dim (int, optional): [description]. Defaults to 128.
+        sn_eps ([type], optional): [description]. Defaults to 1e-6.
+        init_type (str, optional): [description]. Defaults to 'ortho'.
+        split_noise (bool, optional): [description]. Defaults to True.
+        act_cfg ([type], optional): [description]. Defaults to
+            dict(type='ReLU').
+        conv_cfg ([type], optional): [description]. Defaults to
+            dict(type='Conv2d').
+        upsample_cfg ([type], optional): [description]. Defaults to
+            dict(type='nearest', scale_factor=2).
+        with_spectral_norm (bool, optional): [description]. Defaults to
+            True.
+        auto_sync_bn (bool, optional): [description]. Defaults to True.
+        blocks_cfg ([type], optional): [description]. Defaults to
+            dict(type='BigGANGenResBlock').
+        norm_cfg ([type], optional): [description]. Defaults to None.
+        arch_cfg ([type], optional): [description]. Defaults to None.
+        out_norm_cfg ([type], optional): [description]. Defaults to
+            dict(type='BN').
+        pretrained ([type], optional): [description]. Defaults to None.
+    """
+
+    def __init__(self,
+                 output_scale,
+                 noise_size=120,
+                 num_classes=0,
+                 out_channels=3,
+                 base_channels=96,
+                 input_scale=4,
+                 with_shared_embedding=True,
+                 shared_dim=128,
+                 sn_eps=1e-6,
+                 init_type='ortho',
+                 split_noise=True,
+                 act_cfg=dict(type='ReLU'),
+                 conv_cfg=dict(type='Conv2d'),
+                 upsample_cfg=dict(type='nearest', scale_factor=2),
+                 with_spectral_norm=True,
+                 auto_sync_bn=True,
+                 blocks_cfg=dict(type='BigGANGenResBlock'),
+                 norm_cfg=None,
+                 arch_cfg=None,
+                 out_norm_cfg=dict(type='BN'),
+                 pretrained=None,
+                 **kwargs):
+        super().__init__()
+        self.noise_size = noise_size
+        self.num_classes = num_classes
+        self.shared_dim = shared_dim
+        self.with_shared_embedding = with_shared_embedding
+        self.output_scale = output_scale
+        self.arch = arch_cfg if arch_cfg else self._get_default_arch_cfg(
+            self.output_scale, base_channels)
+        self.input_scale = input_scale
+        self.split_noise = split_noise
+        self.blocks_cfg = deepcopy(blocks_cfg)
+        self.upsample_cfg = deepcopy(upsample_cfg)
+
+        # Validity Check
+        # If 'num_classes' equals to zero, we shall set 'with_shared_embedding'
+        # to False and
```

Comment:
Well, in default settings, if users set `num_classes` to zero, they should set `split_noise` to True since we use `cbn`. But if users do not use `cbn`,  `split_noise` may be False. I am wondering how to write the comments.